src/transformers/modeling_gpt2.py

```diff
@@ -489,6 +490,74 @@ def __init__(self, config):
 
         self.init_weights()
 
+        # Model parallel
+        self.model_parallel = False
+        self.device_map = None
+
+    def parallelize(self, device_map):
+        '''Uses a device map to distribute attention modules of the model across several devices.
+        Args: 
+            device_map (:obj:`Dict[int, list]`):
+                A dictionary that maps attention modules to devices. Note that the embedding module and LMHead are
+                always automatically mapped to the first device (for esoteric reasons). That means that the first
+                device should have fewer attention modules mapped to it than other devices. Here is an example of
+                a device map on a machine with 4 GPUs using gpt2-xl, which has a total of 48 attention modules:
+                
+                model = GPT2LMHeadModel.from_pretrained('gpt2-xl')
+                device_map = {0: [0, 1, 2, 3, 4, 5, 6, 7, 8],
+                              1: [9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21],
+                              2: [22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34],
+                              3: [35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47]}
+                model.parallelize(device_map)
+
+                For reference, the gpt2 models have the following number of attention modules:
+                    gpt2: 12
+                    gpt2-medium: 24
+                    gpt2-large: 36
+                    gpt2-xl: 48
+        '''
+        # Check validity of device_map
+        assert self.device_map or device_map, "No device_map detected. Pass a device_map dictionary to parallelize_model"
+        analyze_device_map(device_map, len(self.h))
```

Comment:
(nit) maybe the function name `assert_device_map(...)` is better here? Took me a bit to understand that it's just a checking function.

Reply:
Oh, yes that's much better.